src/Microsoft.Identity.Web.TokenAcquisition/DefaultAuthorizationHeaderProvider.cs

```diff
@@ -113,5 +121,14 @@ private static TokenAcquisitionOptions CreateTokenAcquisitionOptionsFromApiOptio
                 FmiPath = downstreamApiOptions?.AcquireTokenOptions.FmiPath,
             };
         }
+
+        /// <summary>
+        /// Since AcquireTokenOptions is recreated, we need to update the original TokenAcquisitionOptions wth the parameters that were
+        /// updated in the new TokenAcquisitionOptions.
+        /// </summary>
+        private void UpdateOriginalTokenAcquisitionOptions(AcquireTokenOptions? acquireTokenOptions, TokenAcquisitionOptions newTokenAcquisitionOptions)
+        {
+            acquireTokenOptions!.LongRunningWebApiSessionKey = newTokenAcquisitionOptions.LongRunningWebApiSessionKey;
```

Comment:
this is unsafe, we can't say not null here (!) and maybe null where this is called (?)

Reply:
Yes it is, updated